modules/ROOT/pages/salesforce/salesforce-connector-97.adoc

```diff
@@ -0,0 +1,1274 @@
+
+[source,xml,linenums]
+----
+<flow name="flowStreamingChannelReplay">
+    <!-- INBOUND ENDPOINT -->
+    <sfdc:replay-streaming-channel streamingChannel="/u/Notifications" replayId="1" replayOption="ALL"/>
+    <!-- REST OF YOUR FLOW -->
+    <logger level="INFO" message="Replayed events: #[payload]"/>
+</flow>
+----
+
+If the `ALL` or `ONLY_NEW` replay options are selected, then the `replayId` value is ignored.
+
+=== Push Events to a Streaming Channel
+
+Salesforce lets you push custom events to a specific streaming channel through the REST API. You can use the Salesforce https://workbench.developerforce.com/about.php[Workbench] or this connector.
+
+To use `push-generic-event` operation:
+
+[source,xml,linenums]
+----
+<flow name="flowPushGenericEvent">
+    <!-- INBOUND ENDPOINT -->
+    <sfdc:push-generic-event channelId="0M6j0000000KyjBCAS">
+    	<sfdc:events>
+            <sfdc:event payload="Notification message text"/>
+        </sfdc:events>
+	</sfdc:push-generic-event>
+    <logger level="INFO" message="Replayed events: #[payload]"/>
+</flow>
+----
+
+The channel ID can be retrieved from the response map of the `publish-streaming-channel` operation.
+
+Another way of retrieving the ID of the channel is from the Salesforce page, as follows:
+
+. Log into your Developer Edition organization.
+. Under All Tabs (+), select Streaming Channels.
+
+If the channel ID field is not visible on the channel list, then:
+
+. Click Create New View.
+. Type a name for the view in the Name input field.
+. In the Available Fields list, select Streaming Channel ID, and click Add.
+. Add any other fields you want.
+. Click Save.
+
+Now you should see the channel ID for each streaming channel in the list.
+
+The JSON received as response from the push event operation looks something like:
+
+[source,json,linenums]
+----
+[
+	{
+	"userOnlineStatus": {
+	},
+	"fanoutCount": 0
+	}
+]
+----
+
+== Load Data in Batches
+
+The Salesforce Bulk API loads batches of your organization's data into Salesforce.
+
+The Salesforce connector provides the Create and Create Bulk operations for working
+with the Bulk API.
+
+For all bulk operations, Salesforce handles the creation process in the background, so the connector does not reply with a collection of SaveResults because it does not have them yet. Instead, the connector replies with a BatchInfo object, which contains the ID of the batch and the ID of the job it creates to upload those objects.
+
+=== Track the Status of Bulk Data
+
+You can monitor a Bulk API batch in Salesforce through the Job ID for the Bulk Data Load Jobs.
+
+The job detail page in Salesforce includes a related list of all the batches for the job. The related list provides View Request and View Response links for each batch. If a batch is a CSV file, the links return the request or response in CSV format. If a batch is an XML file, the links return the request or response in XML format.

```

Comment:
"all batches for the job"

Reply:
This is a Salesforce term: see https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/asynch_api_intro.htm -- "Bulk API is based on REST principles and is optimized for loading or deleting large sets of data. You can use it to query, queryAll, insert, update, upsert, or delete many records asynchronously by submitting **batches**. Salesforce processes batches in the background."